nautobot/docs/development/core/settings.md

```diff
@@ -0,0 +1,82 @@
+# Adding and Updating Nautobot Settings
+
+Best practices when adding, updating, and deprecating/removing Nautobot configuration settings.
+
+## Consider Constance
+
+If the setting is one that would be desirable to configure at run time rather than install time, and the nature of the setting is such that it *can* be changed at run time without requiring a server reload or similar operations to fully take effect, consider implementing it as a [django-constance](https://django-constance.readthedocs.io/en/latest/) setting.
+
+### Constance Advantages
+
+* Constance-enabled settings can be configured through the UI by any user with admin privileges, rather than requiring filesystem access to the Nautobot installation to modify.
+* Constance-enabled settings changes can take effect with zero downtime, as no server restart is required.
+* Constance-enabled settings can still be set in `nautobot_config.py` as a way to "lock" the configuration and disable the ability for admin users to modify it through the UI.
+
+### Constance Drawbacks
+
+* Constance-enabled settings must be definable through the UI, so advanced settings patterns like callable functions and other Python objects can't generally be implemented through Constance.
+* Many native Django settings can't be implemented through Constance as they typically require a process restart to fully take effect.
+* Care needs to be taken when implementing environment variable support for settings that are also Constance-enabled - you must check for `"<variable_name>" in os.environ` before setting a variable in the Nautobot settings, so that it remains unset (and therefore manageable by Constance) if the environment variable isn't specified, rather than the simpler common pattern of just `SETTING = os.getenv("<variable_name>", <default value>)` used elsewhere in the settings.
+
+## Environment Variable Support
+
+For all settings where it's feasible to do so, you should make sure that it's configurable via environment variable as an alternative to directly editing `nautobot_config.py`. Exceptions would be for settings that cannot easily be defined as an environment variable string, such as Python functions, complex dicts, etc.
+
+The environment variable should typically be named `NAUTOBOT_<setting_name>` unless there is a *strong* existing convention in other tools for a different name.
+
+For Constance-enabled settings, the general pattern would be:
+
+```python
```

Comment:
```suggestion
```python title="nautobot/core/settings.py"
```

It isn't mentioned in this section that this config is meant for that file so this may help?